### PR TITLE
Add Kotlin docs to Auth, Realtime and Storage

### PR DESCRIPTION
**Pull request by jan-tennert**
## What kind of change does this PR introduce?

Docs Update

## What is the current behavior?

There are only JS (and sometimes Dart) snippets for the code snippets in the docs.

## What is the new behavior?
270312277-1d79b067-1b28-4e65-825f-3ab396744ad4
270298271-23e6844d-c64b-458b-9997-73bbc206d45c

There are new snippets and guides for Kotlin:
- All Auth related guides have now Kotlin tabs including the Social Logins. There is also an entry for Google & Apple Native Sign in.
- Realtime Changes, Presence & Broadcast have Kotlin tabs.
- Storage (resumable) uploading/downloading

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-password-reset.mdx [hunk 1]
```diff
@@ -85,13 +171,75 @@ const next = url.searchParams.get('next')
...
+```
+
+```kotlin
+//MainActivity.kt
+override fun onCreate() {
+	client.handleDeeplinks(intent)
+}
+```
+
+That's it! Supabase-KT will take care of parsing and importing the session. Other targets are similar, you can check [OAuth and OTP link verification](https://supabase.com/docs/reference/kotlin/initializing) for more information.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/auth-password-reset.mdx:208]:
```suggestion
That's it! `supabase-kt` will take care of parsing and importing the session. Other targets are similar, you can check [OAuth and OTP link verification](https://supabase.com/docs/reference/kotlin/initializing) for more information.
```

**Comment by kiwicopple** [comment]
looks like it just needs some prettier! Can you run the command `npm run format`?

**Comment by jan-tennert** [comment]
> looks like it just needs some prettier! Can you run the command `npm run format`?

Should be good now.

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks 💚

**Pull request merged**